docs: Align comments in .ort.yml templates

tsteenbe opened the pull request:
Make comments in templates clear and concise and remove mentions of 'product' as not every SW project is a product.

Signed-off-by: Thomas Steenbergen <thomas.steenbergen@here.com>

sschuberth commented:
BTW, I was recently thinking about inlining all these files to [Configuration.md](https://github.com/heremaps/oss-review-toolkit/blob/master/docs/Configuration.md), mostly because I find the file names to be weird (e.g. `bower.yml` sounds like a configuration file for Bower, not for ORT; we could rename it e.g. to `.ort.bower.yml`), but also seeing all these examples at once would give a better overview.

Opinions?

sschuberth commented:
About the actual change proposed, I have to admit I like the original wording with full sentences a tad better.

mnonnenmacher commented:
> About the actual change proposed, I have to admit I like the original wording with full sentences a tad better.

I would also prefer full sentences.

tsteenbe commented:
I have updated the PR to better include why I changed the "comment" - trying to better accommodate non-technical readers. They may not know the finer details of how .ort.yml or scopes work - all they need to understand why this was excluded and that that excluded dependencies are not in released artifacts.

Do not want to inline these templates into Configuration.md - for supporting dev teams using ORT it great to simply give them a direct link to a .ort.yml file that they can download and use as a started for setting up excludes for their project.

Added an new commit to rename file extensions to `.ort.yml` ...

fviernau commented on docs/examples/bower.yml:
```diff
@@ -2,4 +2,4 @@ excludes:
   - name: "devDependencies"
     reason: "BUILD_TOOL_OF"
-    comment: "These are dependencies only used for development. They are not distributed in the context of this product."
+    comment: "Scope with dependencies only used for development. Not included in released artifacts in the context of this project."
```
optional: I personally dislike the redundancy introduced by 'Not included in released artifacts in the context of this project.' because not being distributed is what excludes are ALL ABOUT.

this can be considered out of scope of this change.

tsteenbe replied:
Yes, you and I known that but persons who:
- stumble upon a .ort.yml file in a repository and do not know ORT will have a better chance of understanding what this file is about.
-  are non-technical readers of ORT scan report will instantly know how to interpret  excludes. Concept of excludes is not explained in the reports. Adding `Not included in released artifacts in the context of this project.` is a short-term workaround until In the long-term fix in the reports to explain excludes.

fviernau commented on docs/examples/bower.yml:
```diff
@@ -2,4 +2,4 @@ excludes:
   - name: "devDependencies"
     reason: "BUILD_TOOL_OF"
-    comment: "These are dependencies only used for development. They are not distributed in the context of this product."
+    comment: "Scope with dependencies only used for development. Not included in released artifacts in the context of this project."
```
optional: "Scope with" is also redundant.

I'd like it short: 

comment: "Development dependencies."

tsteenbe replied:
Added "Scope" for the ORT reports for non-technical users - fine with making it shorter to just "Dependencies only used for development"
